File: Codeforces/2240/A/try-1.py
/*
 * Problem : Another Popcount Problem
 * Difficulty: Easy
 * Submission: Try 1
 * status: Accepted
 * Language: Python 3
 * Date: 7/6/2026, 7:01:58 PM
 * Link: https://codeforces.com/contest/2240/problem/A
 */

t = int(input())

for _ in range(t):
    n, k = map(int, input().split())

    ans = 0

    for i in range(31):
        cost = 1 << i
        take = min(n // cost, k)
        ans += take
        n -= take * cost

    print(ans)

File: LeetCode/Medium/11-container-with-most-water/try-1.java
/*
 * Problem #11: Container With Most Water
 * Difficulty: Medium
 * Submission: Try 1
 * status: Accepted
 * Language: java
 * Date: 8/6/2026, 7:39:21 PM
 * Link: https://leetcode.com/problems/container-with-most-water/
 */

class Solution {
    public int maxArea(int[] height) {

        int left = 0;
        int right = height.length - 1;
        int max = 0;

        while (left < right) {

            int h = Math.min(height[left], height[right]);
            int w = right - left;
            int area = h * w;

            max = Math.max(max, area);

            if (height[left] < height[right]) {
                left++;
            } else {
                right--;
            }
        }

        return max;
    }
}
